akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/PersistentFsmToTypedMigrationCompileOnlyTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.akka.persistence.typed;
+
+import akka.actor.typed.ActorRef;
+import akka.persistence.typed.*;
+import akka.persistence.typed.javadsl.*;
+
+import static akka.persistence.fsm.AbstractPersistentFSMTest.WebStoreCustomerFSM.*;
+
+public class PersistentFsmToTypedMigrationCompileOnlyTest {
+
+  //#commands
+  interface Command {}
+
+  public static class AddItem implements Command {
+    public final Item item;
+
+    public AddItem(Item item) {
+      this.item = item;
+    }
+  }
+
+  public static class GetCurrentCart implements Command, ExpectingReply<ShoppingCart> {
+    private final ActorRef<ShoppingCart> replyTo;
+
+    public GetCurrentCart(ActorRef<ShoppingCart> replyTo) {
+      this.replyTo = replyTo;
+    }
+
+    @Override
+    public ActorRef<ShoppingCart> replyTo() {
+      return replyTo;
+    }
+  }
+
+  public enum Buy implements Command {
+    INSTANCE
+  }
+
+  public enum Leave implements Command {
+    INSTANCE
+  }
+  //#commands
+
+  //#state
+  abstract static class State {
+    public final ShoppingCart cart;
+
+    protected State(ShoppingCart cart) {
+      this.cart = cart;
+    }
+  }
+
+  public static class LookingAround extends State {
+    public LookingAround(ShoppingCart cart) {
+      super(cart);
+    }
+  }
+
+  public static class Shopping extends State {
+    public Shopping(ShoppingCart cart) {
+      super(cart);
+    }
+  }
+
+  public static class Inactive extends State {
+    public Inactive(ShoppingCart cart) {
+      super(cart);
+    }
+  }
+
+  public static class Paid extends State {
+    public Paid(ShoppingCart cart) {
+      super(cart);
+    }
+  }
+  //#state
+
+  // #event-adapter
+  public static class PersistentFSMEventAdapter extends EventAdapter<DomainEvent, Object> {
+
+    @Override
+    public Object toJournal(DomainEvent domainEvent) {
+      // leave events as is, can't roll back to PersistentFSM
+      return domainEvent;
+    }
+
+    @Override
+    public String manifest(DomainEvent event) {
+      return "";
+    }
+
+    @Override
+    public EventSeq<DomainEvent> fromJournal(Object event, String manifest) {
+      if (event instanceof StateChangeEvent) {
```

Comment:
one important thing for the deprecation is that we shouldn't deprecate `StateChangeEvent`, since that is needed for migration and possibly always needed for migrated applications later. #26490

Reply:
Added a comment to make sure I don't forget not to deprecate